src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head = null;
+    private Node<T> tail = null;
+    private int size;
```

Comment:
The `Node` class should be an inner class placed at the end of the `MyLinkedList` class. Additionally, the `Node` class should not have getters and setters since we can access its private fields directly from the outer class.